### PR TITLE
Update palette neutral border tokens

### PR DESCRIPTION
**Pull request by lilyvc**
This PR updates palette neutral border tokens (used by container border) as part of Card changes, as well as updating Tabs and AppHeader to use separable border in place of container border.

These changes come from design and are reflected in Figma.

[Tabs Light theme](https://www.figma.com/file/7KgRuAqrlNM8pYSMRthbjt/branch/gI8phnS918i78IAtu3vWr8/Salt-Light-Components?type=design&node-id=7613%3A78321&mode=design&t=ujZ4hO5B4OUSsqri-1) has been updated in branch to reflect token updates - now using separable-secondary-borderColor. Same with [Tabs Dark theme](https://www.figma.com/file/14t5w7BuWParQh6AAnVP0C/branch/glNhIPuNo8rnZebgZI3G23/Salt-Dark-Components?type=design&node-id=5457%3A78514&mode=design&t=qKAnUBGGpYz6nhGb-1). 
 [Tabs 2 component specs](https://www.figma.com/file/wh4iwNOuVIpebbV1qgSnxP/branch/NWVjXtz5Q9EpZ74nDnzCxL/Tabs-2?type=design&node-id=801%3A15352&mode=design&t=3Lp2cfdzTeqHRQOI-1) have also been updated in a branch too.
 [App header pattern characteristics spec](https://www.figma.com/file/nIajdHWJVg7G1ZDCSNx42H/branch/eOOmzdmBDoJzpmv9ATZUNI/App-Header-Pattern?type=design&node-id=41%3A15488&mode=design&t=BVRJAh0sy1nph7a5-1)

### CODE REVIEW

File: packages/theme/css/palette/neutral.css [hunk 1]
```diff
@@ -5,12 +5,12 @@
   --salt-palette-neutral-primary-foreground: var(--salt-color-gray-900);
   --salt-palette-neutral-primary-foreground-disabled: var(--salt-color-gray-900-fade-foreground);
   --salt-palette-neutral-primary-separator: var(--salt-color-black-fade-separatorOpacity-primary);
-  --salt-palette-neutral-primary-border: var(--salt-color-gray-60);
+  --salt-palette-neutral-primary-border: var(--salt-color-gray-40);
   --salt-palette-neutral-primary-border-disabled: var(--salt-color-gray-60-fade-border);
```
Comment (joshwooding) [inline packages/theme/css/palette/neutral.css:9]: Are the other tokens not changing? cc @shey-v

Comment (shey-v) [reply packages/theme/css/palette/neutral.css:9]: Yes, they will be an alpha of 40%.

Comment (lilyvc) [reply packages/theme/css/palette/neutral.css]: updated

**Review by joshwooding — APPROVED** [review]

**Review by feliciackh — DISMISSED** [review]

**Review by lukeac123 — COMMENTED** [review]
chromatic baselines need accepting, otherwise looks good :)

File: packages/core/stories/patterns/app-header/app-header.stories.tsx [hunk 2]
```diff
@@ -61,7 +61,7 @@ export const AppHeader = () => {
                 inset: "0 0 auto 0",
                 paddingRight: "1em",
                 borderBottom:
-                  "var(--salt-size-border) var(--salt-container-borderStyle) var(--salt-container-secondary-borderColor)",
```
[the remark this answers is not in the document]
Comment (lilyvc) [reply packages/core/stories/patterns/app-header/app-header.stories.tsx:64]: Container is still being used for Card, Tabs and App Header are now using separable.

**Review by feliciackh — APPROVED** [review]

**Review by Fercas123 — APPROVED** [review]

**Review by origami-z — COMMENTED** [review]
Given the impact of visual regression, changelog should also touch which components are impacted?

**Pull request merged**